core/station/impl/src/models/request_operation.rs

```diff
@@ -178,10 +179,54 @@ pub struct EditAccountOperation {
     pub input: EditAccountOperationInput,
 }
 
+#[storable]
+#[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub enum ChangeAssets {
+    ReplaceWith {
+        assets: Vec<AssetId>,
+    },
+    Change {
+        add_assets: Vec<AssetId>,
+        remove_assets: Vec<AssetId>,
+    },
+}
+
+impl ChangeAssets {
+    pub fn apply(&self, assets: &mut Vec<AccountAsset>) {
+        match self {
+            ChangeAssets::ReplaceWith { assets: new_assets } => {
+                *assets = new_assets
+                    .iter()
+                    .map(|asset_id| AccountAsset {
+                        asset_id: *asset_id,
+                        balance: None,
+                    })
+                    .collect();
+            }
+            ChangeAssets::Change {
+                add_assets,
+                remove_assets,
+            } => {
+                for asset_id in add_assets {
+                    if !assets.iter().any(|a| a.asset_id == *asset_id) {
```

Comment:
nit: you could create a `Map<AssetId, Asset>` of that list of assets to make the loop more efficient